src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,177 @@
-     *                  if `someDate` is in the future relating to the `current date`;
-     *     - "`someDate` is before `currentDate`"
-     *                  if `someDate` is in the past relating to the `current date`;
-     *     - "`someDate` is today"
-     *                  if `someDate` is today;
+     * - "`someDate` is after `currentDate`"
+     * if `someDate` is in the future relating to the `current date`;
+     * - "`someDate` is before `currentDate`"
+     * if `someDate` is in the past relating to the `current date`;
+     * - "`someDate` is today"
+     * if `someDate` is today;
      */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        LocalDate currentDate = LocalDate.now();
+        if (someDate.isAfter(currentDate)) {
+            return someDate + " is after " + currentDate;
+        } else if (someDate.isBefore(LocalDate.now())) {
+            return someDate + " is before " + currentDate;
+        } else {
+            return someDate + " is today";
+        }
     }
 
     /**
      * Given a String representation of a date and some timezone,
      * return LocalDateTime in this timezone.
      */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateInString);
+        LocalDateTime localDateTime = zonedDateTime.toLocalDateTime();
+        ZoneId zoneId = ZoneId.of(zone);
+        ZoneOffset zoneOffset = zoneId.getRules().getOffset(localDateTime);
+        return LocalDateTime.ofInstant(Instant.parse(dateInString), zoneOffset);
     }
 
     /**
      * Given some LocalDateTime, return an OffsetDateTime with the local time offset applied
      * (`+02:00` for Ukraine).
-     *
+     * <p>
      * Example: we receive a LocalDateTime with a value `2019-09-06T13:17`.
-     *          We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
-     *          where `+02:00` is the offset for our local timezone.
-     *
+     * We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
+     * where `+02:00` is the offset for our local timezone.
+     * <p>
      * OffsetDateTime is recommended to use for storing date values in a database.
      */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return localTime.atOffset(ZoneOffset.of(TIME_ZONE));
     }
 
     /**
      * Given a String formatted as `yyyymmdd`,
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        return (Integer.parseInt(date.substring(START_MONTH_POSITION, END_MONTH_POSITION))
+                > MAX_MONTH) ? Optional.empty()
+                : Optional.ofNullable(LocalDate.from(DateTimeFormatter.BASIC_ISO_DATE.parse(date)));
```

Comment:
Let's not check each specific case when the String format of date would be wrong. Let's just catch DateTimeParseException and return Optional.empty() in this case.

Reply:
done